kratos/containers/distributed_csr_matrix.h

```diff
@@ -760,22 +761,84 @@ class DistributedCsrMatrix final
                 }
             }
             p_csr_output->FinalizeAssemble();
-        }        
-        
+        }
+
         return p_csr_output;
     }
 
+    double NormDiagonal()
+    {
+        // const double diagonal_norm = IndexPartition<IndexType>(size1()).template for_each<SumReduction<double>>([&](IndexType Index) {
```

Comment:
i would implement it here. It is not difficult since you can call the corresponding one of the diagonal blocks and simply reduce the results

Reply:
Done.